File: homework_4/Task3.java
public class Task3
{
    public static void main(String[] args) 
    {
        
        int n = 3; // EX: Matrix size
        int[][] x = 
        {              // INDEX of rows below:
            {1, 0, 0}, // row 0
            {0, 0, 0}, // row 1
            {0, 0, 0}  // row 2
        };

            boolean allZero = false; // flag for checking for all zero

            for (int i = 0; i < n; i++)
            {
                allZero = true; //We can assume a row is all-zero unntil discovered otherwise
                for (int j = 0; j < n; j++)
                {
                    if(x[i][j] != 0) 
                    {
                        allZero = false; // if row has non-zero value flag = false
                        break; //exit
                    }
                }
                if (allZero)
                {
                    System.out.println("First all zero is: " + i);
                    break; //exit for first loop if row is all zero
                }
            }

            if(!allZero)
            {
                System.out.println("No all zero rows.");
            }
    }
}

File: homework_4/Task3.py
def Task3(matrix):
    n = len(matrix) # Dynamic initialize Matrix Size 
    allZero = False

    for i in range(n):
        allZero = True
        for j in range(n):
            if matrix[i][j] != 0:
                allZero = False
                break # exit second loop
        
        if allZero:
            print("First all zero is: ", i)
            break

    if not allZero:
        print("No all zero rows.")


#EX: matrix
x = [
    [0,1,0], # Row 0
    [1,0,1], # Row 1
    [0,0,0]  # Row 2
]

Task3(x) # Runs the function and returns either i = 0, 1, or 2 based off the rows of the matrix from top to bottom. Will print the row index correlating to 
         # which row has all zero and is checked first. 